src/persist-client/src/operators/shard_source.rs

```diff
@@ -125,12 +127,28 @@ where
         Arc::clone(&val_schema),
         should_fetch_part,
     );
+    let (descs, backpressure_token) = {
```

Comment:
is it possible to make this work via a closure passed to shard_source instead? it'd be nice if we could somehow make this all work entirely between shard_source and persist_source, without either of them having to know anything about it

this might also let us relax the `G: Scope<Timestamp = Hybrid<T, u64>>,` on shard_source to be something like a refines bound on `T`

Reply:
@danhhz We could certainly have a closure that takes in the `shard_source_descs` output stream, and needs to out a stream with the same type? I actually like that quite a bit...